cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -279,13 +282,23 @@ private void utilizationForResources(ClusterModel clusterModel, OptimizationOpti
           double capacity = resource.isHostResource() ? broker.host().capacityFor(resource)
                                                       : broker.capacityFor(resource);
           double utilizationPercentage = utilization / capacity;
+          if (utilizationPercentage > lowUtilizationThreshold) {
+            allBrokersBelowLowUtilizationThreshold = false;
+          }
+
           if (utilizationPercentage >= balanceLowerThreshold && utilizationPercentage <= balanceUpperThreshold) {
             numBalancedBrokersInBrokersAllowedReplicaMove++;
           }
           varianceSum += Math.pow(utilization - avgUtilizationPercentage * capacity, 2);
         }
       }
-      _numBalancedBrokersByResource.put(resource, numBalancedBrokersInBrokersAllowedReplicaMove);
+
+      if (allBrokersBelowLowUtilizationThreshold) {
+        // All alive brokers are balanced
+        _numBalancedBrokersByResource.put(resource, clusterModel.aliveBrokers().size());
```

Comment:
Above you are checking whether `_brokersAllowedReplicaMove` are having utilization over `lowUtilizationThreshold` or not. Here you are claiming that _all_ alive brokers are balanced. Which one is it?

Reply:
After offline discussion, this logic is not needed